python/revng/cli/_commands/fetch_debuginfo/pe.py

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env python3
+#
+# This file is distributed under the MIT License. See LICENSE.md for details.
+#
+
+import os
+from pathlib import Path
+
+import pefile
+
+from revng.cli.support import log_error
+
+from .common import download_file, log
+
+
+def fetch_pdb(file: pefile, file_path, urls, verbose):
+    log("Fetching Debugging Information for a PE/COFF...", verbose)
+
+    # Parse the file.
+    try:
+        pe = pefile.PE(file_path)
+    except Exception as exception:
+        log_error("Unable to parse the input file.")
+        log_error(str(exception))
+        return None
+
+    for e in pe.DIRECTORY_ENTRY_DEBUG:
+        if not isinstance(e.entry, pefile.Structure):
+            continue
+        pdb_file_name = e.entry.PdbFileName.decode("utf-8")
+        pdb_file_name = pdb_file_name.replace("\x00", "")
+        # A hash for the PDB file.
+        pdb_id = (
+            str(hex(e.entry.Signature_Data1)[2:])
+            + str(hex(e.entry.Signature_Data2)[2:])
+            + str(hex(e.entry.Signature_Data3)[2:])
+            + e.entry.Signature_Data4.hex()
+            + str(hex(e.entry.Age)[2:])
+        )
+        log("PDBID: " + pdb_id + "...", verbose)
+
+        if "XDG_CACHE_HOME" not in os.environ or os.environ["XDG_CACHE_HOME"] == "":
+            log(
+                "warning: XDG_CACHE_HOME was not set. Using ~/.cache/debuginfod_client/...", verbose
+            )
+            path_to_download = str(Path.home()) + "/.cache/debuginfod_client/" + pdb_id + "/"
+        else:
+            path_to_download = os.environ["XDG_CACHE_HOME"] + "/debuginfod_client/" + pdb_id + "/"
+
+        if not os.path.exists(path_to_download):
+            os.makedirs(path_to_download)
+
+        pdb_file_to_download = path_to_download + "/" + pdb_file_name
+        if os.path.exists(pdb_file_to_download):
```

Comment:
We're trying to migrate away from `os.path` and concatenating strings to [`from pathlib import Path`](https://docs.python.org/3/library/pathlib.html), which is nicer and more portable. Please switch.